projects/packages/publicize/src/class-publicize.php

```diff
@@ -252,7 +252,7 @@ public function get_all_connections_for_user( $args = array() ) {
 								array(
 									'service_name'   => $service_name,
 									'connection_id'  => $connection['connection_data']['id'],
-									'can_disconnect' => self::can_manage_connection( $connection['connection_data'] ),
+									'can_disconnect' => current_user_can( 'edit_others_posts' ) || get_current_user_id() === $user_id,
```

Comment:
I didn't remember making this change, so I didn't add it as a part of https://github.com/Automattic/jetpack/pull/37510

Reply:
That change was made in #37501